sdk/storage/azure-storage-queues/inc/azure/storage/queues/queue_client.hpp

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include <azure/core/credentials/credentials.hpp>
+#include <azure/storage/common/storage_credential.hpp>
+
+#include "azure/storage/queues/queue_options.hpp"
+
+namespace Azure { namespace Storage { namespace Queues {
+
+  class QueueClient final {
+  public:
+    static QueueClient CreateFromConnectionString(
+        const std::string& connectionString,
+        const std::string& queueName,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    explicit QueueClient(
+        const std::string& queueUrl,
+        std::shared_ptr<StorageSharedKeyCredential> credential,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    explicit QueueClient(
+        const std::string& queueUrl,
+        std::shared_ptr<Core::Credentials::TokenCredential> credential,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    explicit QueueClient(
+        const std::string& queueUrl,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    std::string GetUrl() const { return m_queueUrl.GetAbsoluteUrl(); }
+
+    Azure::Response<Models::CreateQueueResult> Create(
+        const CreateQueueOptions& options = CreateQueueOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::CreateQueueResult> CreateIfNotExists(
+        const CreateQueueOptions& options = CreateQueueOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::DeleteQueueResult> Delete(
+        const DeleteQueueOptions& options = DeleteQueueOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::DeleteQueueResult> DeleteIfExists(
+        const DeleteQueueOptions& options = DeleteQueueOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::QueueProperties> GetProperties(
```

Comment:
GetXMethod should return X rather than GetXResult. This is agreed by C++ working group. Ref to the email titled _[SDK model type naming consistency] Client library GetX Methods should return model types named X and not GetXResult_ by Ahson.